### PR TITLE
Get server version from pymysql without additional queries

### PR DESCRIPTION
**Pull request by pasenor**
## Description
We can get rid of two queries that ask the server about it's version: the pymysql connection object already contains everything we want to know.

Also, added a numerical representation of the server version, for future convenience if we start adding any version-specific functionality (which I think we should do).

## Checklist

- [x] I've added this contribution to the `changelog.md`.
- [x] I've added my name to the `AUTHORS` file (or it's already there).

### CODE REVIEW

File: test/test_sqlexecute.py [hunk 1]
```diff
@@ -270,3 +271,19 @@ def test_multiple_results(executor):
...
     assert results == expected
+
+
+@pytest.mark.parametrize(
+    'version_string, species, parsed_version_string, version',
+    (
+        ('5.7.32-35', 'Percona', '35', 35),
+        ('5.7.32-0ubuntu0.18.04.1', 'MySQL', '5.7.32', 50732),
+        ('10.5.8-MariaDB-1:10.5.8+maria~focal', 'MariaDB', '10.5.8', 100508),
+        ('unexpected version string', None, 'unexpected version string', 0),
```
Comment (j-bennet) [inline test/test_sqlexecute.py:282]: This needs a test case for empty string and `None`.

File: mycli/sqlexecute.py [hunk 2]
```diff
@@ -18,17 +20,67 @@
...
+        self.version_str = version_str
+        if species == ServerSpecies.Percona:
+            self.version = int(version_str)
+        else:
+            self.version = self.calc_mysql_version_value(version_str)
+
+    @staticmethod
+    def calc_mysql_version_value(version_str):
+        try:
+            major, minor, patch = version_str.split('.')
```
Comment (j-bennet) [inline mycli/sqlexecute.py:42]: Probably should handle None as well.

File: mycli/main.py [hunk 3]
```diff
@@ -933,7 +933,9 @@ def get_prompt(self, string):
         string = string.replace('\\u', sqlexecute.user or '(none)')
         string = string.replace('\\h', host or '(none)')
         string = string.replace('\\d', sqlexecute.dbname or '(none)')
-        string = string.replace('\\t', sqlexecute.server_type()[0] or 'mycli')
+        string = string.replace(
+            '\\t',
+            sqlexecute.server_info.species and sqlexecute.server_info.species.name or 'mycli')
```
Comment (j-bennet) [inline mycli/main.py:938]: We probably want `mysql`, not `mycli` here?

File: mycli/sqlexecute.py [hunk 4]
```diff
@@ -18,17 +20,67 @@
...
+                species = ServerSpecies.MariaDB
+                parsed_version_str = mysql_version
+            elif comment.isdigit():
+                species = ServerSpecies.Percona
+                parsed_version_str = comment
+            else:
+                species = ServerSpecies.MySQL
+                parsed_version_str = mysql_version
+        else:
+            species = None
```
Comment (j-bennet) [inline mycli/sqlexecute.py:66]: How about adding `ServiceSpecies.Unknown` (or Generic or something like that)? This way, you won't need to check `species` for None.

Comment (pasenor) [reply mycli/sqlexecute.py:66]: good idea

File: mycli/main.py (hunk 3 above)
Comment (pasenor) [reply mycli/main.py:938]: well, that's the current implementation: if we don't understand, say 'mycli' (probably because it may be some renamed fork of mysql)

Comment (j-bennet) [reply mycli/main.py:938]: I think maybe it was a typo from the beginning :)

Comment (pasenor) [reply mycli/main.py:941]: hmm, I get your point, it's kind of inconsistent: if this spot is reserved for the server, we should not replace it with the client name. Would it be appropriate to spell the generic server also `MySQL`?

**Comment by pasenor** [comment]
After some research it seems that special treatment of Percona DB is not warranted since dialect-wise it should be perfectly compatible with the underlying MySQL version.

File: mycli/main.py (hunk 3 above)
Comment (j-bennet) [reply mycli/main.py:941]: I think so. You could set "ServerSpecies.Unknown" to map to string "MySQL", and just use `species.name` here, and never even check for None, like this:

```
string = string.replace('\\t', sqlexecute.server_info.species.name)
```

and in sqlexecute, instead of defaulting `server_info` to None, have this:

```
self.server_info = ServerInfo.from_version_string('unknown')
```

Comment (pasenor) [reply mycli/main.py]: > instead of defaulting server_info to None, have this

server_info is never really None, the code goes like
```Python
def __init__(self, ...):
    self.server_info = None
    self.connect()  # server_info is always set inside connect
```
and initializing it to anything other than None is just a completely wasteful object creation. I guess the only reason it is created in `__init__` is to appease the linter :)
